src/Datagrid/SimplePager.php

```diff
@@ -79,6 +79,11 @@ public function getResults($hydrationMode = null)
         if (\count($this->results) > $this->getMaxPerPage()) {
             $this->haveToPaginate = true;
 
+            @trigger_error(
+                sprintf('Returning a type different than array from %s is deprecated since sonata-project/admin-bundle 3.x and will return array starting from 4.0.', __METHOD__),
+                E_USER_DEPRECATED
+            );
```

Comment:
People are not going to be able to fix this deprecation, are they?

Reply:
Right. The only way to remove this deprecation notice is to update to 4.x

Reply:
Then there should be no deprecation. The goal of a deprecation is to tell people about the upgrade path. Here, there is no upgrade path.

Reply:
Then there should be no deprecation IMO.

Reply:
By the way, it looks like we are already returning an `ArrayCollection`… I don't understand this PR.

Reply:
@greg0ire it all started as https://github.com/sonata-project/SonataAdminBundle/issues/5137 (https://github.com/sonata-project/SonataAdminBundle/pull/5144)

Reply:
This PR is just about adding deprecation to 3.x, because 4.x already returns only `array`

Reply:
I think the deprecation should go anyway, because there is nothing the users can do to make it stop (they can't update to 4.x since there is no stable release).

Reply:
It could stay if there was a way for the user to:
- make the deprecation stop
- make `getResults` consistently return an array on 3.x

Reply:
There is no way to stop deprecation and changing ArrayCollection to array in 3.x would probably be a BC break, so how we can handle this?

Reply:
I would remove the `trigger_error` entirely, and keep the other changes in your PR.

Reply:
Or rather, move them to `UPGRADE-4.0`, which should exist on master (so I would target master)

Reply:
Alternatively, create a new method that always returns arrays, and deprecate the old in favor of the new.

Reply:
But that would imply creating a new interface extending `PagerInterface`, which sounds like too much work already.

Reply:
I've decided to remove this trigger_error.